From 18018992d0e862af47dae855b0aa70faf6689496 Mon Sep 17 00:00:00 2001
From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Fri, 30 Sep 2022 12:04:15 +0200
Subject: [PATCH 85/92] man: add a man page for the VNC backend

Add a new weston-vnc man page, based on the weston-rdp man page.
Also add links in the main weston man page.

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
(cherry picked from commit ad936f1de89ab7306455785fc0f9ebcba2380e1d)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 man/meson.build    |  9 ++++++++
 man/weston-vnc.man | 57 ++++++++++++++++++++++++++++++++++++++++++++++
 man/weston.man     | 12 ++++++++++
 3 files changed, 78 insertions(+)
 create mode 100644 man/weston-vnc.man

diff --git a/man/meson.build b/man/meson.build
index a2b8edc2..5d3b8d8f 100644
--- a/man/meson.build
+++ b/man/meson.build
@@ -53,3 +53,12 @@ if get_option('backend-rdp')
 		configuration: man_conf
 	)
 endif
+
+if get_option('backend-vnc')
+	configure_file(
+		input: 'weston-vnc.man',
+		output: 'weston-vnc.7',
+		install_dir: join_paths(dir_man, 'man7'),
+		configuration: man_conf
+	)
+endif
diff --git a/man/weston-vnc.man b/man/weston-vnc.man
new file mode 100644
index 00000000..64910972
--- /dev/null
+++ b/man/weston-vnc.man
@@ -0,0 +1,57 @@
+.TH WESTON-RDP 7 "2017-12-14" "Weston @version@"
+.SH NAME
+weston-vnc \- the VNC backend for Weston
+.SH SYNOPSIS
+.B weston --backend=vnc-backend.so
+.
+.\" ***************************************************************
+.SH DESCRIPTION
+The VNC backend allows to run a
+.B weston
+environment without the need of specific graphic hardware, or input devices. Users can interact with
+.B weston
+only by connecting using the remote framebuffer protocol (RFB).
+
+The VNC backend uses Neat VNC to implement the VNC part, it acts as a VNC server
+listening for incoming connections. It supports different encodings for encoding
+the graphical content, depending on what is supported by the VNC client.
+
+The VNC backend is not multi-seat aware, so if a second client connects to the
+backend, the first client will be disconnected.
+
+Note that authentication and encryption are not supported yet. Anyone with
+access to the port can get control of the desktop via the VNC output, and
+all data is transferred in plaintext.
+
+.\" ***************************************************************
+.SH CONFIGURATION
+.
+The VNC backend uses the following entries from
+.BR weston.ini .
+.SS Section vnc
+.TP
+\fBrefresh-rate\fR=\fIrate\fR
+Specifies the desktop redraw rate in Hz. If unspecified, the default is 60Hz. Changing
+this may be useful if you have a faster than 60Hz display, or if you want to reduce updates to
+reduce network traffic.
+
+.\" ***************************************************************
+.SH OPTIONS
+.
+When the VNC backend is loaded,
+.B weston
+will understand the following additional command line options.
+.TP
+.B \-\-width\fR=\fIwidth\fR
+The width of the framebuffer. It defaults to 640.
+.TP
+.B \-\-height\fR=\fIheight\fR
+The height of the framebuffer. It defaults to 480.
+.TP
+\fB\-\-port\fR=\fIport\fR
+The TCP port to listen on for connections. It defaults to 5900.
+
+.\" ***************************************************************
+.SH "SEE ALSO"
+.BR weston (1)
+.\".BR weston.ini (5)
diff --git a/man/weston.man b/man/weston.man
index 654c0dae..0c6e5c56 100644
--- a/man/weston.man
+++ b/man/weston.man
@@ -45,6 +45,13 @@ to the desktop is done by using the RDP protocol. Each connecting
 client has its own seat making it a cheap way to test multi-seat support. See
 .BR weston-rdp (7),
 if installed.
+.TP
+.I vnc-backend.so
+The VNC backend runs in memory without the need of graphical hardware. Access
+to the desktop is done by using the RFB protocol. Currently only one
+connecting client is supported. See
+.BR weston-vnc (7),
+if installed.
 .
 .\" ***************************************************************
 .SH SHELLS
@@ -264,6 +271,10 @@ pixman library for software compsiting.
 See
 .BR weston-rdp (7).
 .
+.SS VNC backend options:
+See
+.BR weston-vnc (7).
+.
 .
 .\" ***************************************************************
 .SH FILES
@@ -367,4 +378,5 @@ weston
 .BR weston-debug (1),
 .BR weston-drm (7),
 .BR weston-rdp (7),
+.BR weston-vnc (7),
 .BR weston.ini (5)
-- 
2.20.1

